File: internal/entity/product_entity.go
package entity

import "github.com/google/uuid"

type Product struct {
	ID          string  `json:"id"`
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Price       float64 `json:"price"`
	CategoryID  string  `json:"category_id"`
	ImageURL    string  `json:"image_url"`
}

func NewProduct(name string, description string, price float64, categoryID string, imageUrl string) *Product {
	return &Product{
		ID:          uuid.New().String(),
		Name:        name,
		Price:       price,
		Description: description,
		CategoryID:  categoryID,
		ImageURL:    imageUrl,
	}
}

File: src/products/entities/product.entity.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";

@Entity('products')
export class Product {
   @PrimaryGeneratedColumn('uuid')
   id:          string;

   @Column()
   name:        string;

   @Column({ type: 'text'})
   description: string;

   @Column()
   price:       number;

   @Column()
   image_url:   string;

}
